docs/design/consensus.md

```diff
@@ -1,24 +1,30 @@
 ## Consensus
 
-The consensus protocol in FTLib acts as a shadow precondition for any collective communication operations. Any changes from the consensus protocol will reset the initialization flag of FTLib to `False`, deterring any communication operations after the rebuild procedure returns success.
+The consensus protocol in FTLib acts as a shadow precondition for any collective communication operations. Any changes from the consensus protocol will reset the `_initialized` flag of FTLib to `False`, deterring any communication operations after the rebuild procedure returns success. Under passive or active mode, such change takes place immediately or eventually (when FTLib check the consensus next time).
 
-A member list is maintained by the implementation of consensus protocol.
+A member list is maintained by an implementation of consensus protocol.
 
-The rank-assign scheme in FTLib can extract worker identification from the member list, such like address of each worker. This unique identification helps the rank-assign scheme to designate individual rank number to each worker, which most communication libraries require when initializing.
+The rank-assign scheme in FTLib can extract worker identification from the member list, such like address of each worker, which helps the rank-assign scheme to designate individual rank number to each worker.
```

Comment:
Consider adding a short description on "rank-assign scheme" even though it's pretty intuitive to readers with background

Reply:
Here is how I describe the "rank-assign scheme":

We map the identification of each worker with a hash function and sort the workers with respect to hash value. The worker with minimal hash value will be considered as the *master* worker. While the *master* worker acts identically as other workers in the training phase, it may have additional components or operations during initialization or other stages.